tests/e2e/testcases/icm/sendMsg/suite.go

```diff
@@ -0,0 +1,522 @@
+package sendmsg
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanche-cli/tests/e2e/commands"
+	"github.com/ava-labs/avalanche-cli/tests/e2e/utils"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+const (
+	ewoqKeyName = "ewoq"
+	subnetName  = "testSubnet"
+	subnet2Name = "testSubnet2"
+	cChain      = "cchain"
+	message     = "Hello World"
+)
+
+var _ = ginkgo.Describe("[ICM] sendMsg", func() {
+	ginkgo.BeforeEach(func() {
+		commands.StartNetwork()
+	})
+
+	ginkgo.AfterEach(func() {
+		commands.CleanNetwork()
+		err := utils.DeleteConfigs(subnetName)
+		gomega.Expect(err).Should(gomega.BeNil())
+		utils.DeleteCustomBinary(subnetName)
+
+		err = utils.DeleteConfigs(subnet2Name)
+		gomega.Expect(err).Should(gomega.BeNil())
+		utils.DeleteCustomBinary(subnet2Name)
+	})
+
+	ginkgo.Context("with valid input", func() {
+		ginkgo.It("should send a message from c-chain to subnet", func() {
+			commands.CreateSubnetEvmConfigNonSOV(subnetName, utils.SubnetEvmGenesisPath, false)
+			commands.DeploySubnetLocallyNonSOV(subnetName)
+
+			// Deploy ICM
+			_, err := commands.DeployICMContracts([]string{}, utils.TestFlags{
+				"key":        ewoqKeyName,
+				"blockchain": subnetName,
+			})
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			// Deploy the relayer
+			_, err = commands.DeployRelayer(
+				[]string{
+					"deploy",
+					"--cchain",
+				},
+				utils.TestFlags{
+					"key":           ewoqKeyName,
+					"blockchains":   subnetName,
+					"amount":        10000,
+					"cchain-amount": 10000,
+					"log-level":     "info",
+				})
+			gomega.Expect(err).Should(gomega.BeNil())
+
+			// Send a message
+			globalFlags := utils.GlobalFlags{
+				"local":             true,
+				"skip-update-check": true,
+			}
+
+			sendMsgFlags := utils.TestFlags{
+				"key": ewoqKeyName,
+			}
+
+			sendMessageArgs := []string{
+				cChain,
+				subnetName,
+				message,
+			}
+
+			output, err := utils.TestCommand(utils.ICMCmd, "sendMsg", sendMessageArgs, globalFlags, sendMsgFlags)
+			gomega.Expect(err).Should(gomega.BeNil())
+			gomega.Expect(output).Should(gomega.ContainSubstring(fmt.Sprintf("Delivering message \"%s\" from source blockchain \"%s\"", message, cChain)))
+			gomega.Expect(output).Should(gomega.ContainSubstring(fmt.Sprintf("Waiting for message to be delivered to destination blockchain \"%s\"", subnetName)))
+			gomega.Expect(output).Should(gomega.ContainSubstring("Message successfully Teleported!"))
```

Comment:
I think these three lines can be abstracted into a function

Reply:
Not a fan of exporting assertions. I think some redundancy in tests is okay, as long as it makes them more readable